test/system/src/test/java/io/pravega/test/system/TransactionStressTest.java

```diff
@@ -0,0 +1,390 @@
+                                         .collect(Collectors.toList());
+
+        URI controllerURI = URI.create("tcp://" + String.join(",", uris));
+        log.info("Controller Service direct URI: {}", controllerURI);
+
+        //4.start 3 instances of pravega segmentstore
+        Service segService = new PravegaSegmentStoreService("segmentstore", zkUri, controllerURI);
+        if (!segService.isRunning()) {
+            segService.start(true);
+        }
+        segService.scaleService(3, true);
+        List<URI> segUris = segService.getServiceDetails();
+        log.debug("Pravega Segmentstore service  details: {}", segUris);
+
+    }
+
+    @Before
+    public void setup() {
+
+        // Get zk details to verify if controller, SSS are running
+        Service zkService = new ZookeeperService("zookeeper");
+        List<URI> zkUris = zkService.getServiceDetails();
+        log.debug("Zookeeper service details: {}", zkUris);
+        //get the zk ip details and pass it to  host, controller
+        URI zkUri = zkUris.get(0);
+
+        // Verify controller is running.
+        controllerInstance = new PravegaControllerService("controller", zkUri);
+        assertTrue(controllerInstance.isRunning());
+        List<URI> conURIs = controllerInstance.getServiceDetails();
+        log.info("Pravega Controller service instance details: {}", conURIs);
+
+        // Fetch all the RPC endpoints and construct the client URIs.
+        final List<String> uris = conURIs.stream().filter(uri -> uri.getPort() == 9092).map(URI::getAuthority)
+                                         .collect(Collectors.toList());
+
+        controllerURIDirect = URI.create("tcp://" + String.join(",", uris));
+        log.info("Controller Service direct URI: {}", controllerURIDirect);
+
+        // Verify segment store is running.
+        segmentStoreInstance = new PravegaSegmentStoreService("segmentstore", zkUri, controllerURIDirect);
+        assertTrue(segmentStoreInstance.isRunning());
+        log.info("Pravega Segmentstore service instance details: {}", segmentStoreInstance.getServiceDetails());
+        //executor service
+        executorService = Executors.newFixedThreadPool(NUM_READERS + NUM_WRITERS);
+        //get Controller Uri
+        controller = new ControllerImpl(controllerURIDirect);
+        //read and write count variables
+        eventsReadFromPravega = new ConcurrentLinkedQueue<>();
+        stopReadFlag = new AtomicBoolean(false);
+        eventData = new AtomicLong(0); //data used by each of the writers.
+        eventReadCount = new AtomicLong(0); // used by readers to maintain a count of events.
+        eventWriteCount = new AtomicLong(0);
+    }
+
+    @After
+    public void tearDown() {
+        controllerInstance.scaleService(1, true);
+        segmentStoreInstance.scaleService(1, true);
+        executorService.shutdownNow();
+        eventsReadFromPravega.clear();
+    }
+
+    @Test(timeout = 600000)
```

Comment:
The expected throughput is of (50000 *50 * 8 /*long*/ * 2/*Read and write*/)/10*60 = 66 kbps. If the throughput is this low, it needs fixing and is a test failure.